cunumeric/array.py

```diff
@@ -2053,6 +2087,46 @@ def diagonal(
                 raise ValueError("Either axis1/axis2 or axes must be supplied")
         return self._diag_helper(offset=offset, axes=axes, extract=extract)
 
+    def trace(self, offset=0, axis1=None, axis2=None, dtype=None, out=None):
```

Comment:
Let's add `@add_boilerplate()` here, to handle the conversion of `out` to `cunumeric.ndarray` in case the user calls the `trace` method instead of `cunumeric.trace`.